File: application/backend/service/mobile-app-service/trpo8-mobile-app-service-api/src/main/java/com/imoiseyenko/spastructure/backend/service/mobileapp/MobileAppService.java
package com.imoiseyenko.spastructure.backend.service.mobileapp;

import java.util.List;
import com.imoiseyenko.spastructure.backend.service.mobileapp.exception.MobileAppServiceException;
import com.imoiseyenko.spastructure.backend.service.mobileapp.vo.MobileAppVO;

/**
 * Mobile application service interface.
 * 
 * @author imoiseyenko93@gmail.com
 * @version Dec 19, 2014
 */
public interface MobileAppService {

	/**
	 * Get mobile application with specified id.
	 * 
	 * @param id
	 * @return
	 * @throws MobileAppServiceException
	 * @author imoiseyenko93@gmail.com
	 */
	public MobileAppVO getMobileAppById (Long id) throws MobileAppServiceException;

	/**
	 * Get list of mobile applications.
	 * 
	 * @return
	 * @throws MobileAppServiceException
	 * @author imoiseyenko93@gmail.com
	 */
	public List<MobileAppVO> getAllMobileApps () throws MobileAppServiceException;

	/**
	 * Update mobile application.
	 * 
	 * @param mobileAppVO
	 * @throws MobileAppServiceException
	 * @author imoiseyenko93@gmail.com
	 */
	public void updateMobileApp (MobileAppVO mobileAppVO) throws MobileAppServiceException;

	/**
	 * Delete mobile application with specified id.
	 * 
	 * @param id
	 * @throws MobileAppServiceException
	 * @author imoiseyenko93@gmail.com
	 */
	public void deleteMobileAppById (Long id) throws MobileAppServiceException;
}

File: application/frontend/trpo8-web-client/src/main/webapp/resources/js/app/service/mobileapp/mobileappService.js
/**
 * Mobile application service module.
 */
define(["angular", "app/repository/mobileapp/mobileappRepository"], function (angular, mobileappRepositoryModule) {

	var mobileappServiceModule = angular.module("mobileappServiceModule", ["mobileappRepositoryModule"]);

	mobileappServiceModule.factory("mobileappService", ["mobileappRepository", function (mobileappRepository) {

		return {

			getMobileAppById: function (accessToken, id, successHandler, errorHandler) {

				return mobileappRepository.findMobileAppById(accessToken, id, successHandler, errorHandler);
			},

			getAllMobileApps: function (accessToken, successHandler, errorHandler) {

				return mobileappRepository.findAllMobileApps(accessToken, successHandler, errorHandler);
			},

			updateMobileApp: function (accessToken, mobileApp, successHandler, errorHandler) {

				return mobileappRepository.updateMobileApp(accessToken, mobileApp, successHandler, errorHandler);
			},

			deleteMobileAppById: function (accessToken, id, successHandler, errorHandler) {

				return mobileappRepository.deleteMobileAppById(accessToken, id, successHandler, errorHandler);
			}
		};
	}]);

	return mobileappServiceModule;
});
